crates/ty_server/src/server.rs

```diff
@@ -170,13 +172,80 @@ impl Server {
 
     fn event_loop(
         connection: &Connection,
-        _client_capabilities: &ClientCapabilities,
+        client_capabilities: &ClientCapabilities,
         mut session: Session,
         worker_threads: NonZeroUsize,
     ) -> crate::Result<()> {
         let mut scheduler =
             schedule::Scheduler::new(&mut session, worker_threads, connection.make_sender());
 
+        let fs_watcher = client_capabilities
+            .workspace
+            .as_ref()
+            .and_then(|workspace| workspace.did_change_watched_files?.dynamic_registration)
+            .unwrap_or_default();
+
+        if fs_watcher {
+            let registration = lsp_types::Registration {
+                id: "workspace/didChangeWatchedFiles".to_owned(),
+                method: "workspace/didChangeWatchedFiles".to_owned(),
+                register_options: Some(
+                    serde_json::to_value(DidChangeWatchedFilesRegistrationOptions {
+                        watchers: vec![
+                            FileSystemWatcher {
+                                glob_pattern: lsp_types::GlobPattern::String("**/ty.toml".into()),
```

Comment:
Mostly curious: does this mean that we're reacting to changes that match the `**/ty.toml` pattern. Or that we actively scan the whole project for this pattern when starting the server (which could be quite slow?)

Reply:
The VS code file watcher sends us notifications if any file inside the project matching `ty.toml` gets changed.